core/src/test/java/org/infinispan/expiration/impl/ClusterExpirationFunctionalTest.java

```diff
@@ -313,4 +314,29 @@ public void testMaxIdleAccessSuspectedExpiredEntryRefreshesProperly() {
 
       assertEquals(value, cache0.get(key));
    }
+
+   public void testWriteExpiredEntry() throws InterruptedException {
+      String key = "key";
+      String value = "value";
+
+      for (int i = 0; i < 100; ++i) {
```

Comment:
Why 100 and not just 1?

Reply:
It requires replacing an expired entry - and doesn't always happen since the remove expired is async.

Reply:
Couldn't we use `ControlTimeService` to make it happen deterministically?

Reply:
We have to also coordinate 2 rpcs to interleave at exact points.